src/directLine.ts

```diff
@@ -520,6 +533,49 @@ export class DirectLine implements IBotConnection {
         .catch(error => this.catchPostError(error));
     }
 
+    putActivity(activity: Activity) {
+        // If we're not connected to the bot, get connected
+        // Will throw an error if we are not connected
+        konsole.log("putActivity", activity);
+        return this.checkConnection(true)
+        .flatMap(_ =>
+            Observable.ajax({
+                method: "PUT",
+                url: `${this.domain}/conversations/${this.conversationId}/activities/${activity.id}`,
+                body: activity,
+                timeout,
+                headers: {
+                    "Content-Type": "application/json",
+                    "Authorization": `Bearer ${this.token}`
+                }
+            })
+            .map(ajaxResponse => ajaxResponse.response.id as string)
+            .catch(error => this.catchPostError(error))
+        )
+        .catch(error => this.catchExpiredToken(error));
+    }
+
+    deleteActivity(activityId: string) {
+        // If we're not connected to the bot, get connected
+        // Will throw an error if we are not connected
+        konsole.log("deleteActivity", activityId);
+        return this.checkConnection(true)
+        .flatMap(_ =>
+            Observable.ajax({
+                method: "DELETE",
+                url: `${this.domain}/conversations/${this.conversationId}/activities/${activityId}`,
+                timeout,
+                headers: {
+                    "Content-Type": "application/json",
+                    "Authorization": `Bearer ${this.token}`
+                }
+            })
+            .map(ajaxResponse => ajaxResponse.response.id as string)
+            .catch(error => this.catchPostError(error))
+        )
+        .catch(error => this.catchExpiredToken(error));
+    }
+
```

Comment:
This looks fine from a code perspective, but it's hard to evaluate without understanding the bigger picture. I don't understand who would call these - adaptive cards, I'm guessing?

Also, is there no corresponding `putMessageWithAttachments` for updating messages with well attachments?